src/Sema.zig

```diff
@@ -32094,7 +32094,23 @@ fn analyzeSlice(
                     array_ty = double_child_ty;
                     elem_ty = double_child_ty.childType(mod);
                 } else {
-                    return sema.fail(block, src, "slice of single-item pointer", .{});
+                    if (uncasted_end_opt == .none) {
+                        return sema.fail(block, src, "slice of single-item pointer must have end value of 1", .{});
+                    }
+                    if (!try sema.isComptimeKnown(uncasted_start)) {
+                        return sema.fail(block, start_src, "slice of single-item pointer must have comptime-known start", .{});
+                    }
+                    if (!try sema.isComptimeKnown(uncasted_end_opt)) {
+                        return sema.fail(block, end_src, "slice of single-item pointer must have comptime-known end", .{});
+                    }
```

Comment:
It would be nice if you could combine these errors to a single one which explains that the slice must be `[0..1]`.

Reply:
I can combine the error messages, though before I can finalise an error message, I would need an answer to https://github.com/ziglang/zig/pull/17944#issuecomment-1804983411. If `ptr[0..0]` is allowed it might be a bit tricky to make a nice message that covers all cases.